plugins/rest/src/plugin.cpp

```diff
@@ -0,0 +1,86 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2022 The VAST Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include "rest/authenticator.hpp"
+#include "rest/configuration.hpp"
+#include "rest/generate_token_command.hpp"
+#include "rest/server_command.hpp"
+#include "rest/specification_command.hpp"
+
+#include <vast/concept/convertible/data.hpp>
+#include <vast/concept/convertible/to.hpp>
+#include <vast/concept/printable/to_string.hpp>
+#include <vast/data.hpp>
+#include <vast/plugin.hpp>
+#include <vast/system/node.hpp>
+#include <vast/system/node_control.hpp>
+#include <vast/system/spawn_or_connect_to_node.hpp>
+#include <vast/type.hpp>
+
+#include <caf/scoped_actor.hpp>
+#include <caf/typed_event_based_actor.hpp>
+
+namespace vast::plugins::rest {
+
+/// The API plugin.
+class plugin final : public virtual command_plugin,
+                     public virtual component_plugin {
+  /// Initializes a plugin with its respective entries from the YAML config
+  /// file, i.e., `plugin.<NAME>`.
+  /// @param config The relevant subsection of the configuration.
+  [[nodiscard]] caf::error initialize(data data) override {
+    // FIXME
+    auto config = to<configuration>(data);
+    return {};
+  }
+
+  /// Returns the unique name of the plugin.
+  [[nodiscard]] const char* name() const override {
+    return "rest";
```

Comment:
In terms of naming, do you think it makes sense to call this `web` instead, since it's more powerful than just a REST API?

Reply:
Yes, makes sense to me